pyro/infer/importance.py

```diff
@@ -44,3 +44,33 @@ def _traces(self, *args, **kwargs):
                 poutine.replay(self.model, trace=guide_trace)).get_trace(*args, **kwargs)
             log_weight = model_trace.log_prob_sum() - guide_trace.log_prob_sum()
             yield (model_trace, log_weight)
+
+    def get_log_normalizer(self):
+        """
+        Estimator of the normalizing constant of the target distribution.
+        (mean of the unnormalized weights)
+        """
+        log_w = torch.tensor(self.log_weights)
+        log_num_samples = torch.log(torch.tensor(self.num_samples, dtype=torch.float))
+        return torch.logsumexp(log_w - log_num_samples, 0)
+
+    def get_normalized_weights(self, log_scale=False):
+        """
+        Compute the normalized importance weights.
+        """
+        log_w = torch.tensor(self.log_weights)
+        log_w_norm = log_w - torch.logsumexp(log_w, 0)
+        return log_w_norm if log_scale else torch.exp(log_w_norm)
+
+    def get_ESS(self):
+        """
+        Compute (Importance Sampling) Effective Sample Size (ESS).
+        """
+        # check to make sure list is not empty
+        if self.log_weights:
+            log_w_norm = self.get_normalized_weights(log_scale=True)
+            ess = torch.exp(-torch.logsumexp(2*log_w_norm, 0))
+        else:
+            logger.warn("The log_weights list is empty, effective sample size is zero.")
```

Comment:
can you `import warnings` up top and use `warnings.warn()` here? (and also above though not in the scope of this PR)

Reply:
Sure. Will also fix the other one.